aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -1322,6 +1322,46 @@ Tensor outer(const Tensor& self, const Tensor& vec2) {
             TYPE, NAME, __VA_ARGS__)
 #endif
 
+
+#ifdef __aarch64__
+static inline int64_t get_mkldnn_matmul_min_dim() {
+  static auto value = [&] {
+    // Minimum dimension requirement for MKLDNN; derived based on experiments.
+    constexpr int64_t default_min_dim = 8;
```

Comment:
Hmm, why not allow users to specify min_dim via environment variable on x86 as well?

Reply:
@malfet I think there are two reasons we want to keep such environment variables specific to aarch64:

1. We're trying to improve performance on aarch64, and all the analysis and benchmarking were done on aarch64 (AWS Graviton c7g). So it makes more sense not to expose such variables for x86 because that wouldn't be meaningful.
2. The other reason is it appears to me MKL on x86 doesn't go through this [mkldnn_matmul](https://github.com/pytorch/pytorch/blob/29813c61ea61d616230d3251ade81f56472ecb77/aten/src/ATen/native/mkldnn/Matmul.cpp#L121) code path (i.e. doesn't go through ideep->oneDNN). At least from the hugging-face "bert-base-uncased" model it's what I have observed. Therefore, such heuristics probably won't make sense for x86.

Reply:
@yd2102 just to clarify: I'm not proposing to change the default to 8 and 8k on x86, I just want to avoid code bifurcation by having the same options available across all platforms, that also keeps divergent codepath smaller (i.e. you would need 2 line divergence to set default `min_dim` rather than 20+ lines divergence you are having right now)

Reply:
@malfet I have pushed the change to converge the code paths based on your suggestion.